src/backend/joanie/core/models/products.py

```diff
@@ -511,6 +548,8 @@ def save(self, *args, **kwargs):
 
             try:
                 # orders with free product are automatically validated
+                if self.total == 0.0 and self.state == enums.ORDER_STATE_DRAFT:
+                    self.submit()
```

Comment:
submit should never be done automatically as it is the customer's move.
Only validation can be automatic in case price is zero but it should not be done in the save method, it should be done in the submit transition.